public/overview/index.php

```diff
@@ -1,6 +1,13 @@
 <?php
+// Import required dependencies
+include_once '../../config.php';
 include '../../db.php';
 
+// Regenerate session ID after login or logout
+if (isset($_POST['logout']) || (!isset($_SESSION['masterpass']) && !isset($_SESSION['username']))) {
```

Comment:
How does this fire when the user is logged in?

Reply:
it doesn't. removed it from the comment.